### PR TITLE
Support Python 3.5+

### PR DESCRIPTION
**Pull request by blakerosenthal**
Add support for Python3 with Python2 backwards compatibility.

Major changes in this commit:
- Update `setup.py` to require `activitysim >= 0.9.1` and also bump other package versions to match activitysim's dependencies
- Correct syntax in `*.yml` files to be compatible with latest `pyyaml` version (5.1)
- Replace discontinued `activitysim.core.inject_defaults` import module with `activitysim.core.config`
  - use `inject` instead of `orca` for injectables
  - use `activitysim.core.inject`'s `.add_table(..., ..., replace=True)` for pipeline repop cases to explicitly replace injected tables
- Replace old Python2 code (print statements, division, absolute imports, etc.) with Python3 code and ensure backwards compatibility with `__future__` module
- Also fix several future warnings with old `pandas` code (`.ix` to `.loc`, etc)
- Update `.travis.yml` to also run tests with Python3.4/5/6

Resolves https://github.com/RSGInc/populationsim/issues/69

### CODE REVIEW

**Comment by blakerosenthal** [comment]
> Maybe consider replacing old_div calls with // if that is python2 compatible in some way?

@toliwaga Initial tests that replace `old_div` with `//` seem to break the tests. Something about `//` in Python 3 is different than `/` in Python 2. I'll read some documentation and see if I can spot the difference.

**Comment by blakerosenthal** [comment]
@toliwaga Removed calls to old_div! had to ensure proper data types in a few locations (floats divide differently than ints in py2 vs. py3). Should be good for a final look! Let me know what you think.

File: docs/getting_started.rst [hunk 1]
```diff
@@ -51,12 +54,34 @@ Installation
...
+  <https://activitysim.github.io/activitysim/gettingstarted.html#anaconda>`__ guide.
+
+
 Run Examples
 ------------
 
   * Before running examples, ensure that Anaconda Python, dependent libraries and PopulationSim package have been installed.
 
-  * Download and unzip the `example setups <https://github.com/RSGInc/populationSim_resources/raw/master/example_setup/PopulationSimExampleSetUps.7z>`_ to a folder on your computer. It does not have to be the same directory as your Anaconda or PopulationSim install.
+  * Download and unzip the `example setups <https://github.com/RSGInc/populationSim_resources/raw/master/example_setup/PopulationSimExampleSetUpsPython3.zip>`_ to a folder on your computer. It does not have to be the same directory as your Anaconda or PopulationSim install.
```
Comment (blakerosenthal) [inline docs/getting_started.rst]: this folder has already been updated with the latest Python3 compatible tests

Comment (toliwaga) [reply docs/getting_started.rst]: This looks good

**Comment by toliwaga** [comment]
> example_weighting replaces example_test_no_integerizing?

I see that my test folder example_test_no_integerizing was never checked in 

so example_weighing is test code for no-integerization version for weighting?

**Comment by blakerosenthal** [comment]
> so example_weighing is test code for no-integerization version for weighting?

@toliwaga that's what it looks like. Could you check in your `example_test_no_integerizing` folder in a new PR?

**Comment by blakerosenthal** [comment]
> maybe document test_full_run2_repop_replace to the effect taht:
> # tests are run in alphabetical order
> # and we expect to find the pipeline file from test_full_run1 in output

I can do that!

**Pull request merged**